### ISSUE TITLE
Expandable/collapsible pattern implementation that centers a11y

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2287]
Current implementations of expandable/collapsible content do not meet accessibility standards, assessed in their usage for value annotations and linked resource property accordions (as well as URI Dereferencer trigger links, which are discussed in [a module-specific issue](https://github.com/omeka-s-modules/UriDereferencer/issues/18)). An accessible expand/collapse pattern should be implemented that could carry over to other text components, as has been requested for Hierarchy ([omeka-s-modules/Hierarchy#11](https://github.com/omeka-s-modules/Hierarchy/issues/11)).

For the current pattern (tested on [value annotations on this page](https://dev.omeka.org/amayer/amayer-s/omeka-s/s/my-omeka-s-site/item/24030) and [linked resource accordions on this page](https://dev.omeka.org/amayer/amayer-s/omeka-s/s/my-second-site/item-set/4147)) which applies Lively, upon expanding or collapsing trigger links, focus is lost in the state change, moving to the web content area. This happens immediately after activation for value annotations, while the accordion panel allows one to expand and collapse a trigger while maintaining focus, but focus does move to the web content area once a user attempts to tab away. For accordion items, obscured elements are receiving focus, which moves into the collapsed containers and through all associated elements before a user can tab to the next accordion item (fails [SC 2.4.11](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html)). Accessible names are present for value annotations but missing the visible text labels assigned to the links (fails [SC 2.5.3](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html)). Accessible descriptions might also come into play here to better describe what function is to be expected.

What should happen:
- Names (and maybe descriptions as well) are read out for trigger links that update according to state change
    - Text or accessible names are unique when multiple trigger links are present
- Trigger links are able to be activated via Space or Enter
- Page maintains logical focus order that is not disturbed by activating expand or collapse
- Focus either moves into expanded content when link is activated, or remains at original trigger element
    - Expanded content can then be tabbed to as well as away from
- Focus remains on original trigger element upon collapse
- Page positioning does not jump when content is expanded/collapsed
- Focus does not move to obscured elements (e.g. collapsed containers that do not expand when focus is moved into them)
- Searching the page brings up text hidden behind collapse elements (does not currently happen via page search or VO search) without causing obscured focus

It appears that value annotation links do not employ `aria-expanded` attributes, while linked resource accordions and URI dereferencer trigger links do. Why is this? Harvard’s Digital Accessibility guide provides [an example of expandable sections](https://accessibility.huit.harvard.edu/technique-expandable-sections); their implementation also includes `aria-hidden` attributes, which are not present in the current implementations, alongside `aria-expanded`. I imagine this could be contributing to the obscured focus issue.

It is important to the note that I have tested with VoiceOver, which can have some limitations with `aria-expanded` (as can JAWS). Thus, overall support needs to be considered in implementation.
